\documentclass{beamer}
\usetheme{metropolis}
\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{graphicx}
\usepackage{array,booktabs,tabularx}
\usepackage{epstopdf}
\usepackage{colortbl, xcolor}


\title{signalHsmm: prediction of malarial signal peptides}
\date{}
\author{Micha\l{} Burdukiewicz\inst{1}*, Piotr Sobczyk\inst{2}, Pawe\l{} B\l{}a\.{z}ej\inst{1}, Pawe\l{} Mackiewicz\inst{1}}
\institute{\textsuperscript{1}University of Wroc\l{}aw, Department of Genomics, 

\textsuperscript{2}Wroc\l{}aw University of Science and Technology, Faculty of Pure and Applied Mathematics,
}

\begin{document}
  \maketitle
  \section{Signal peptides}
  
\begin{frame}{}
<<knitrIntro, echo = FALSE, message=FALSE,warning=FALSE>>=
source("my_ggplot_theme.R")

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

library(dplyr)
library(reshape2)
library(xtable)
library(biogram)

load("./dat/signalhsmm_poster.RData")

freq_nondeg <- read.csv2("./dat/freq_nondeg.csv")[, -1]
freq_deg <- read.csv2("./dat/freq_deg.csv")[, -1]

do_pca <- function(x) 
  x %>% 
  select(-type, -taxon) %>% 
  prcomp(center = TRUE, scale = TRUE) %>% 
  getElement("x") %>% 
  data.frame() %>% 
  select(1, 2) %>% 
  cbind(select(x, type, taxon), .) %>% 
  mutate(type_nice = factor(type, labels = c("Mature peptide", "Signal peptide")),
         taxon_nice = factor(taxon, labels = c("other eukaryotes", "Plasmodium"))) %>% 
  mutate(both = paste0(type_nice, "\n(", taxon_nice, ")\n")) %>% 
  filter(type_nice == "Signal peptide")

dat_deg <- do_pca(freq_deg) 
dat_nondeg <- do_pca(freq_nondeg)

# dat_deg <- read.table("PCAgr.txt", header = TRUE, sep = "\t")
# dat_nondeg <- read.table("PCA.txt", header = TRUE, sep = "\t")
# colnames(dat_deg) <- c("both", "PC1", "PC2")
# colnames(dat_nondeg) <- c("both", "PC1", "PC2")

plot_pca <- function(x)
  ggplot(x, aes(x = PC1, y = PC2, fill = both)) + 
  geom_density_2d(color = "black", contour = TRUE) +
  #geom_point() +
  stat_density2d(aes(fill=both,alpha=..level..), color = "black", contour = TRUE, geom="polygon") +
  scale_linetype_discrete("") +
  scale_fill_manual("", values = c("dodgerblue1", "firebrick1"), 
                    labels = c("Other eukaryotes", "Plasmodium sp.")) +
  scale_shape_discrete("") +
  scale_color_discrete("") +
  scale_x_continuous("Principal Component 1\n") +
  scale_y_continuous("Principal Component 2") +
  scale_alpha_continuous(range = c(0.25, 0.4)) +
  guides(alpha = FALSE) +
  my_theme

p1 <- plot_pca(dat_nondeg) 
p2 <- plot_pca(dat_deg)
leg <- g_legend(p1)


@

Secretory signal peptides:
        \begin{itemize}[<+->]
          \item are short (20-30 residues) N-terminal amino acid sequences forming $\alpha$-helices,
            \item direct a protein to the endomembrane system and next to extra- or intracellular localizations,
            \item are universal enough to direct properly proteins in different secretory systems; artifically introduced bacterial signal peptides can guide proteins in mammals~\citep{nagano_2014} and plants~\citep{moeller_2009},
            \item tag hormons, immune system proteins, structural proteins, and metabolic enzymes.
        \end{itemize}

\end{frame}

  
  \begin{frame}{Architecture}

  \begin{figure}
\includegraphics[width=0.9\textwidth]{static_figure/SP.png}
\end{figure}
Signal peptides possess three distinct domains with variable length and characteristic amino acid composition~\citep{hegde_surprising_2006}:
      \begin{itemize}
        \item n-region: mostly basic residues~\citep{nielsen_prediction_1998},
        \item h-region: strongly hydrophobic residues~\citep{nielsen_prediction_1998},
        \item c-region: a few polar, uncharged residues~\citep{jain_signal_1994}.
      \end{itemize}
\end{frame}




% 1. download plasmodium genomes, show bias
% 2. how bias alters amino acid composition? show mutations
% 3. amino acid frequencies in a) eukarytic signal peptides and plasmodium signal peptides


\section{Malarial signal peptides}

\begin{frame}{Apicoplast}
Four membrane-bounded plastid of Plasmodium sp. responsible for several biochemical pathways including the biosynthesis of fatty acids, isoprenoids and haem.


\begin{figure}
\includegraphics[width=0.7\textwidth]{static_figure/apicoplast_tem.png}
\end{figure}

\citet{ralph_tropical_2004}
\end{frame}


\begin{frame}{Bi-partite transit signal}
The \textbf{signal peptide} is required for targeting the apicoplast protein to the endomembrane system, whereas the \textbf{transit peptide} is required to traffic the protein to the apicoplast.

\begin{figure}
\includegraphics[width=0.9\textwidth]{static_figure/apicoplast.jpg}
\end{figure}

\citet{kalanon_malaria_2010}
\end{frame}

\begin{frame}{Apicoplast}
The absence of a metabolic counterpart in human host make the apicoplast proteins promising targets for anti-malarial drug development.

\end{frame}

\begin{frame}
Heavy adenine-thymine bias of malarial genomes alters amino acid composition of malarial signal peptides making them hard to predict using software trained on other eukaryotes.

<<echo = FALSE,message=FALSE,results='asis',fig.width=14, fig.height=8>>=
mfreq <- filter(freq_nondeg, type == "signal") %>% 
  melt(variable.name = "residue", value.name = "freq") %>% 
  mutate(taxon = factor(taxon, labels = c("other", "Plasmodium")))

plas_freq <- group_by(mfreq, taxon, residue) %>% 
  summarise(med = median(freq)) %>% 
  dcast(residue ~ taxon) %>% 
  mutate(plas = ifelse(Plasmodium - other > 0, "plas", ifelse(Plasmodium - other == 0, "eq", "other"))) %>% 
  select(residue, plas) %>% 
  mutate(plas = factor(plas, labels = c("Rare amino acids", "Typical for other\nsignal peptides", 
                                        "Typical for Plasmodium sp.\nsignal peptides")))


ggplot(inner_join(mfreq, plas_freq), aes(x = residue, y = freq, color = taxon)) +
  geom_boxplot(position = position_dodge(width = 0.9), notch = TRUE) +
  facet_wrap(~ plas, ncol = 3, scales = "free_x") +
  scale_x_discrete("Amino acid") +
  scale_y_continuous("Normalized frequency") +
  scale_color_manual("Taxon", values = c("dodgerblue1", "firebrick1"), 
                     labels = c("Other eukaryotes", "Plasmodium sp.")) +
  my_theme
@
\small If notches are overlapping, two groups can be considered equal.

\end{frame}

\begin{frame}{PCA of amino acid frequency}
<<echo = FALSE,message=FALSE,results='asis',fig.width=14, fig.height=9>>=
print(p1)
@ 
\end{frame}

\begin{frame}
Since amino acid composition of signal peptides differ between Plasmodium sp. and other eukaryotes, predictors of signal peptides do not detect malarial signal peptides accurately.

There are no enough malarial signal peptides to train a specialized predictor.
  \end{frame}



\begin{frame}{Aim}
  Can we employ decision rules used for prediction of eukaryotic signal peptides to correctly detect malarial signal peptides?
  \end{frame}

\begin{frame}
Even nonbiological sequences can be effective signal peptides provided they fulfill general requirements~\citep{tonkin_evolution_2008}.

$\textnormal{NH}_2$-SKINNYSLINKYKINKYTHING-COOH - targets apicoplast.

$\textnormal{NH}_2$-ITWILLNEVERTARGETPLASTID-COOH - does not target apicoplast.

\end{frame}


\section{Methods}

\begin{frame}{Reduced amino acid alphabets}
To date, several reduced amino acid alphabets have been proposed, which have been applied to (among others) protein folding and protein structure prediction.
  \end{frame}

\begin{frame}{Novel reduced amino acid alphabets}
13 physicochemical properties handpicked from AAIndex database relevant to the regional architecture of signal peptides.
\tiny
  \begin{table}[ht]
\begin{tabular}{ll}
  \toprule
Property name & Amino acid scale \\ 
  \midrule
Size & Size \\ 
   \rowcolor[gray]{0.85}Size & Molecular weight\\ 
  Size & Residue volume \\ 
   \rowcolor[gray]{0.85}Size & Bulkiness \\ 
  Hydrophobicity & Normalized hydrophobicity scales for $\alpha$-proteins \\ 
   \rowcolor[gray]{0.85}Hydrophobicity & Consensus normalized hydrophobicity scale\\ 
  Hydrophobicity & Hydropathy index \\ 
   \rowcolor[gray]{0.85}Hydrophobicity & Surrounding hydrophobicity in $\alpha$-helix\\ 
  Polarity & Polarity \\ 
   \rowcolor[gray]{0.85}Polarity & Mean polarity \\ 
  Occurrence in $\alpha$-helices & Signal sequence helical potential \\ 
   \rowcolor[gray]{0.85}Occurrence in $\alpha$-helices & Normalized frequency of N-terminal helix\\ 
  Occurrence in $\alpha$-helices & Relative frequency in $\alpha$-helix \\ 
   \bottomrule
\end{tabular}
\end{table}
\end{frame}

\begin{frame}{Cross-validation}
We built 96 reduced amino acid alphabets (each based on one scale per a given property category) of length 4 (four distinct regions: n-, h-, c-region, mature protein).

Alphabets were evaluated in a cross-validation experiment using hidden semi-Markov models trained on eukaryotic sequences.
\end{frame}

\section{Results}

\begin{frame}{Cross-validation}

<<echo = FALSE,message=FALSE,results='asis',fig.width=14, fig.height=9>>=
cv_dat <- read.csv("./dat/cv_res.csv") %>% 
  mutate(best = Sens == max(Sens))

ggplot(cv_dat, aes(x = Sens, y = Spec, color = AUC, shape = best)) +
  geom_point(size = 5) +
  scale_color_gradient(low = "blue", high = "red") +
  scale_x_continuous("Sensitivity") +
  scale_y_continuous("Specificity") +
  scale_shape_discrete("Best sensitivity reduced alphabet") +
  guides(color = guide_colorbar(barwidth = unit(50, "line"))) +
  my_theme
@


\end{frame}

\begin{frame}
\begin{table}[ht]
\centering
\begin{tabular}{ll}
  \toprule
Group & Amino acids \\ 
  \midrule
I & D, E, H, K, N, Q, R \\ 
   \rowcolor[gray]{0.85}II & G, P, S, T, Y \\ 
  III & F, I, L, M, V, W \\ 
   \rowcolor[gray]{0.85} IV & A, C \\ 
   \bottomrule
\end{tabular}
\end{table}

\end{frame}

\begin{frame}
\begin{table}[ht]
\centering
\begin{tabular}{ll}
  \toprule
Group & Amino acids \\ 
  \midrule
\rowcolor{red!20}I & D, E, H, K, N, Q, R \\ 
   II & G, P, S, T, Y \\ 
  III & F, I, L, M, V, W \\ 
   IV & A, C \\ 
   \bottomrule
\end{tabular}
\end{table}

I. Charged or uncharged but polar amino acids absent in h-region.
\end{frame}

\begin{frame}
\begin{table}[ht]
\centering
\begin{tabular}{ll}
  \toprule
Group & Amino acids \\ 
  \midrule
I & D, E, H, K, N, Q, R \\ 
   \rowcolor{red!20}II & G, P, S, T, Y \\ 
  III & F, I, L, M, V, W \\ 
   IV & A, C \\ 
   \bottomrule
\end{tabular}
\end{table}

II. Polar and uncharged amino acids common in c-region.
\end{frame}


\begin{frame}
\begin{table}[ht]
\centering
\begin{tabular}{ll}
  \toprule
Group & Amino acids \\ 
  \midrule
I & D, E, H, K, N, Q, R \\ 
   II & G, P, S, T, Y \\ 
  \rowcolor{red!20}III & F, I, L, M, V, W \\ 
   IV & A, C \\ 
   \bottomrule
\end{tabular}
\end{table}

III. Hydrophobic amino acids common in h-region.
\end{frame}

\begin{frame}{PCA of amino acid frequency}
<<echo = FALSE,message=FALSE,results='asis',fig.width=14, fig.height=9>>=
print(p2)
@ 


Signal peptides, after the reduction of the amino acid alphabet, group together despite their different origins.

\end{frame}



\begin{frame}{PCA of amino acid frequency}
<<echo = FALSE,message=FALSE,results='asis',fig.width=14, fig.height=11>>=
grid.arrange(p1 + guides(fill = FALSE) + ggtitle("Full alphabet"), 
             p2 + guides(fill = FALSE) + ggtitle("Reduced alphabet") , 
             leg, nrow = 3, heights = c(0.48, 0.48, 0.04))

@ 

\end{frame}

\begin{frame}{Benchmark}
Benchmark data set: 51 proteins with signal peptide and 211 proteins without signal peptide from members of \textit{Plasmodiidae}.
\end{frame}
    
\begin{frame}{Benchmark}    
    \begin{table}

\centering
\begin{tabular}{rll}
\hline
                                                                              & MCC               & AUC               \\ \hline
signalP 4.1 (no tm) \citep{petersen_signalp_2011}                         & 0.6872            & 0.8667            \\
\rowcolor[gray]{0.85}signalP 4.1 (tm) \citep{petersen_signalp_2011} & 0.6196            & 0.7951            \\
signalP 3.0 (NN) \citep{DyrlovBendtsen2004783}                              & 0.7220            & 0.8938            \\
\rowcolor[gray]{0.85}signalP 3.0 (HMM) \citep{DyrlovBendtsen2004783}  & 0.5553            & 0.7734            \\
%PrediSi \citep{hiller_predisi:_2004}                                      & 0.3849            & 0.6453            \\
%\rowcolor[gray]{0.85}Philius \citep{reynolds_transmembrane_2008}    & 0.5684            & 0.7707            \\
Phobius \citep{kall_combined_2004}                                        & 0.5895            & 0.7880            \\
\hline
\hline
\rowcolor[gray]{0.85}signalHsmm-2010                                    & 0.7409            & 0.9262            \\
signalHsmm-2010 (hom. 50\%)                                                   & \textbf{0.7621} & \textbf{0.9384} \\
\rowcolor[gray]{0.85}signalHsmm-2010 (full alphabet)                           & 0.6853            & 0.8718            \\
\hline
\end{tabular}
\end{table}

signalHsmm-2010: trained on data set of 3676 eukaryotic proteins with signal peptides added before year 2010.
    
\end{frame}


\begin{frame}{Conclusions and perspectives}

Eukaryotic signal peptides have very similar amino acid composition in their regions considering only the physicochemical properties of residues.

signalHsmm allows sensitive scanning of malarial proteome for potential drug targets.

\end{frame}


\begin{frame}{Availability}

signalHsmm web-server \url{http://smorfland.uni.wroc.pl/shiny/signalHsmm}.

signalHsmm R package \url{https://CRAN.R-project.org/package=signalHsmm}.

\end{frame}


\begin{frame}{Acknowledgements and funding}

This research was partially funded by the KNOW Consortium and National Science Center (2015/17/N/NZ2/01845).

\small{
\begin{itemize}
\item Paweł Mackiewicz.
\item \textbf{biogram} package (\url{https://cran.r-project.org/package=biogram}):
\begin{itemize}
\small
\item Piotr Sobczyk,
\item Chris Lauber.
\end{itemize}
\end{itemize}
}

\end{frame}



\begin{frame}[allowframebreaks]
        \frametitle{References}
  \bibliographystyle{apalike}
  \bibliography{signalhsmm_poster}
\end{frame}


\end{document}